docs/route-tasks.md

```diff
@@ -0,0 +1,138 @@
+
+## Use Task Queues
+
+"Task Routing" is the development technique that makes use of Task Queues to orchestrate the execution of Tasks by specific Workers.
+
+Let's look at the ["Hello World" sample in Go](https://github.com/temporalio/temporal-go-samples/tree/master/helloworld) to see how this is done.
+
+Here, we are creating a new Worker that is subscribed to the "hello-world" Task Queue, and is registered to handle the `helloworld.Workflow` and `helloworld.Activity`:
+
+```go
+// The Worker is created and subscribed to the "hello-world" Task Queue
+w := worker.New(c, "hello-world", worker.Options{})
+defer w.Stop()
+// The Worker is registered to handle the helloworld Workflow
+w.RegisterWorkflow(helloworld.Workflow)
+// The Worker is registered to handle the helloworld Activity
+w.RegisterActivity(helloworld.Activity)
+```
+
+And here, we are starting the `helloworld.Workflow`:
+
+```go
+// The Workflow Options specify the "hello-world" Task Queue
+// This is the Task Queue that the Workflow will add its Tasks to
+workflowOptions := client.StartWorkflowOptions{
+  ID:        "hello_world_workflowID",
+  TaskQueue: "hello-world",
+}
+// The Workflow Options are passed to the execution of the Workflow
+we, err := c.ExecuteWorkflow(
+  context.Background(), 
+  workflowOptions, 
+  helloworld.Workflow, 
+  "Temporal",
+)
+```
+
+Task routing can be as simple as making sure a Worker is subscribed to a Task Queue and Tasks are being added to it.
+
+:::caution
+
+If a Worker pulls a Task that it has not been registered to run, an "unknown workflow type" error will occur.
+
+:::
+
+## Task routing for Activity dependencies
+
+The Temporal service will persist Activity results as long as the results can be serialized and stored in a Cassandra or MySQL database (which happens to be true for most Activity results). However, there are some scenarios where an Activity execution will do something that results in data being stored on the host where the Worker is running. It is also possible that the next Activity in the Workflow has a dependency on that data.
+
+Let's look at a file processing Workflow, as an example,  where files are downloaded, processed in some way, and then uploaded to some other location. If a large video file is being processed only the name of the file can persist normally. The file itself must persist locally on the Worker host. This means that any future Activities that act on the video file must also take place on the host where the Worker downloaded it.
+
+So, how do we ensure that a Worker on the correct host will execute the next set of Activities that depends on the video file?
+
+Each SDK offers a slightly different way to handle this.
+
+### Java
+
+In Java the paradigm is to create two Workers. Each will exist on the same host and thus have access to the same file system, but one will listen to a shared Task Queue, which it will poll to start the Workflow, and the other will listen to its own private Task Queue.
```

Comment:
"paradigm" meh - best practice maybe

Reply:
"exist" isn't the right word.

Also the "poll to start the Workflow" part isn't super easy to grasp

Reply:
> "paradigm" meh - best practice maybe

paradigm (noun): a typical example or pattern of something; a model.
best practice (noun): commercial or professional procedures that are accepted or prescribed as being correct or most effective.

It is much more of a pattern/model - there is really no other alternative to make this happen. Paradigm is by far a better word choice here IMO. Best practice suggests there a bunch of ways to do it - but through experience it is considered the better way... which just is not the case here. This is the pattern that Temporal forces a developer to adhere to -